src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2580,6 +2582,14 @@ PruneShardList(Oid relationId, Index tableId, List *whereClauseList,
 	Var *partitionColumn = PartitionColumn(relationId, tableId);
 	char partitionMethod = PartitionMethod(relationId);
 
+	/* short circuit for reference tables */
+	if (partitionMethod == DISTRIBUTE_BY_NONE)
+	{
+		Assert(list_length(shardIntervalList) == 1);
```

Comment:
> I feel like we tend to overuse Assert.

Hmm, that makes sense. I mostly try to add many `Asserts` during the development and remove on the PR. 

For this PR, I'll consider removing some of the `Asserts` I added.